flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/deserializer/PulsarDeserializationSchema.java

```diff
@@ -19,29 +19,25 @@
 package org.apache.flink.connector.pulsar.source.reader.deserializer;
 
 import org.apache.flink.annotation.PublicEvolving;
-import org.apache.flink.api.common.ExecutionConfig;
 import org.apache.flink.api.common.serialization.DeserializationSchema;
 import org.apache.flink.api.common.serialization.DeserializationSchema.InitializationContext;
-import org.apache.flink.api.common.typeinfo.TypeInformation;
 import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
-import org.apache.flink.connector.pulsar.common.schema.PulsarSchema;
-import org.apache.flink.connector.pulsar.source.PulsarSourceBuilder;
 import org.apache.flink.connector.pulsar.source.config.SourceConfiguration;
 import org.apache.flink.util.Collector;
 
 import org.apache.pulsar.client.api.Message;
 import org.apache.pulsar.client.api.Schema;
-import org.apache.pulsar.common.schema.KeyValue;
 
 import java.io.Serializable;
 
 /**
  * A schema bridge for deserializing the pulsar's {@code Message<byte[]>} into a flink managed
- * instance. We support both the pulsar's self managed schema and flink managed schema.
+ * instance. We support both the pulsar's {@link Schema} and flink's {@link DeserializationSchema}.
  *
  * @param <T> The output message type for sinking to downstream flink operator.
  */
 @PublicEvolving
+@SuppressWarnings("java:S112")
```

Comment:
Why to add this suppression?

Reply:
This is used for SonarQube plugin in IntelliJ IDEA. Sonar require a specified exception instead of a generic one.